packages/ses/src/compartment-shim.js

```diff
@@ -72,7 +101,45 @@ export class Compartment {
 
     assign(globalObject, endowments);
 
+    // Map<FullSpecifier, ModuleCompartmentRecord>
+    const moduleRecords = new Map();
+    // Map<FullSpecifier, ModuleInstance>
+    const instances = new Map();
+    // Map<FullSpecifier, Alias{Compartment, FullSpecifier}>
+    const aliases = new Map();
+    // Map<FullSpecifier, {ExportsProxy, ProxiedExports, activate()}>
+    const deferredExports = new Map();
+
+    for (const [specifier, module] of entries(modules)) {
+      if (typeof module === 'string') {
+        throw new TypeError(
+          `Cannot map module ${q(specifier)} to ${q(
+            module,
+          )} in parent compartment`,
+        );
+      } else {
+        const alias = moduleAliases.get(module);
+        if (alias != null) {
+          // Modules from other components.
+          aliases.set(specifier, alias);
+        } else {
+          // TODO create and link a synthetic module instance from the given namespace object.
```

Comment:
How convinced are you that we do eventually want to do this TODO? I am not clear how to think about it.

Reply:
I’m pretty sure we’ll need this to be able to inject initial capability-bearing modules like `fs`. The alternative is to brew these modules in an endowed compartment with a single source file that converts the properties of the endowment into exported names, which would be ceremony for ceremony’s sake.